packages/imperative/src/events/src/EventProcessor.ts

```diff
@@ -0,0 +1,164 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { Logger } from "../../logger/src/Logger";
+import { EventCallback, EventTypes } from "./EventConstants";
+import { ImperativeError } from "../../error/src/ImperativeError";
+import { Event } from "./Event";
+import { ConfigUtils } from "../../config/src/ConfigUtils";
+import { LoggerManager } from "../../logger/src/LoggerManager";
+import { ImperativeConfig } from "../../utilities";
+import { EventUtils } from "./EventUtils";
+import { IEventDisposable } from "./doc";
+import { IProcessorTypes } from "./doc/IEventInstanceTypes";
+
+/**
+ * Manages event subscriptions and emissions for a specific application.
+ *
+ * @export
+ * @class EventProcessor
+ */
+export class EventProcessor {
+    public subscribedEvents: Map<string, Event> = new Map();
+    public eventTimes: Map<string, string>;
+    public processorType: IProcessorTypes;
+    public appName: string;
+    public logger: Logger;
+
+    /**
+     * Constructor initializes a new instance of EventProcessor.
+     *
+     * @param {string} appName - The application's name.
+     * @param {IProcessorTypes} type - The type of processor (Emitter, Watcher, or Both).
+     * @param {Logger} [logger] - Optional logger for recording events and errors.
+     */
+    public constructor(appName: string, type: IProcessorTypes, logger?: Logger) {
+        EventUtils.validateAppName(appName);
+
+        this.subscribedEvents = new Map();
+        this.eventTimes = new Map();
+        this.appName = appName;
+        this.processorType = type;
+
+        // Ensure we have correct environmental conditions to setup a custom logger,
+        // otherwise use default logger
+        if (ImperativeConfig.instance.loadedConfig == null || LoggerManager.instance.isLoggerInit === false) {
+            ConfigUtils.initImpUtils("zowe");
+        }
+        this.logger = logger ?? Logger.getImperativeLogger();
+    }
+
+    /**
+     * Subscribes to shared events by creating and managing a new subscription.
+     *
+     * @param {string} eventName - The name of the event to subscribe to.
+     * @param {EventCallback[] | EventCallback} callbacks - Callback functions to handle the event.
+     * @returns {IEventDisposable} - Object allowing management of the subscription.
+     */
+    public subscribeShared(eventName: string, callbacks: EventCallback[] | EventCallback): IEventDisposable {
+        if (this.processorType === IProcessorTypes.EMITTER) {
+            throw new ImperativeError({ msg: `Processor does not have correct permissions: ${eventName}` });
+        }
```

Comment:
Should we add `@throws` to the typedocs for `subscribeShared` and `subscribeUser`?